sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/HyperLogLogPlusPlusHelper.scala

```diff
@@ -93,6 +94,8 @@ class HyperLogLogPlusPlusHelper(relativeSD: Double) extends Serializable {
     val value = dataType match {
       case FloatType => FLOAT_NORMALIZER.apply(_value)
       case DoubleType => DOUBLE_NORMALIZER.apply(_value)
+      case st: StringType if !st.supportsBinaryEquality && _value.isInstanceOf[UTF8String] =>
```

Comment:
```suggestion
      case st: StringType if !st.supportsBinaryEquality =>
```

Reply:
StringType values must be `UTF8String`. The entire the SQL execution framework takes this assumption.